opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -171,30 +172,30 @@ def __init__(
         self._metric_readers = metric_readers
 
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader._register_measurement_consumer(self)
+            metric_reader._set_collect_callback(
+                self._measurement_consumer.collect
+            )
 
+        self._shutdown_once = Once()
         self._shutdown = False
 
     def force_flush(self) -> bool:
 
         # FIXME implement a timeout
 
-        metric_reader_result = True
-
         for metric_reader in self._sdk_config.metric_readers:
-            metric_reader_result = (
-                metric_reader_result and metric_reader.force_flush()
-            )
-
-        if not metric_reader_result:
-            _logger.warning("Unable to force flush all metric readers")
-
-        return metric_reader_result
+            metric_reader.collect()
+        return True
```

Comment:
Is this supposed to be `True` regardless of what happens with `metric_reader.collect`?

Reply:
`collect` doesn't return anything, so returning True until we implement timeout/error handling mechanism seemed reasonable to me. What do you think?

Reply:
All right, opened #2402 to track this.